tests/unit/s2n_kem_test.c

```diff
@@ -113,346 +113,398 @@ int main(int argc, char **argv)
-    };
-    {
-        /* Happy case for s2n_kem_recv_public_key() */
-        struct s2n_kem_params kem_params = { .kem = &s2n_test_kem };
-
-        DEFER_CLEANUP(struct s2n_blob io_blob = { 0 }, s2n_free);
-        EXPECT_SUCCESS(s2n_alloc(&io_blob, TEST_PUBLIC_KEY_LENGTH + 2));
-        struct s2n_stuffer io_stuffer = { 0 };
-        EXPECT_SUCCESS(s2n_stuffer_init(&io_stuffer, &io_blob));
-
-        /* {0, 2} = length of public key to follow
-         * {2, 2} = test public key */
-        const uint8_t input[] = { 0, 2, 2, 2 };
-        EXPECT_SUCCESS(s2n_stuffer_write_bytes(&io_stuffer, input, TEST_PUBLIC_KEY_LENGTH + 2));
-        EXPECT_SUCCESS(s2n_stuffer_reread(&io_stuffer));
-
-        EXPECT_SUCCESS(s2n_kem_recv_public_key(&io_stuffer, &kem_params));
-
-        /* s2n_kem_recv_public_key() should alloc kem_params->public_key and nothing else */
-        EXPECT_EQUAL(kem_params.public_key.size, TEST_PUBLIC_KEY_LENGTH);
-        EXPECT_NOT_EQUAL(0, kem_params.public_key.allocated);
-        EXPECT_BYTEARRAY_EQUAL(kem_params.public_key.data, TEST_PUBLIC_KEY, TEST_PUBLIC_KEY_LENGTH);
-        EXPECT_EQUAL(0, kem_params.shared_secret.allocated);
-        EXPECT_EQUAL(0, kem_params.private_key.allocated);
-        EXPECT_SUCCESS(s2n_kem_free(&kem_params));
-    };
-    {
-        /* Failure cases for s2n_kem_recv_public_key() */
-        EXPECT_FAILURE_WITH_ERRNO(s2n_kem_recv_public_key(NULL, NULL), S2N_ERR_NULL);
-
-        DEFER_CLEANUP(struct s2n_blob io_blob = { 0 }, s2n_free);
-        EXPECT_SUCCESS(s2n_alloc(&io_blob, 1));
-        struct s2n_stuffer io_stuffer = { 0 };
-        EXPECT_SUCCESS(s2n_stuffer_init(&io_stuffer, &io_blob));
-
-        EXPECT_FAILURE_WITH_ERRNO(s2n_kem_recv_public_key(&io_stuffer, NULL), S2N_ERR_NULL);
-
-        struct s2n_kem_params kem_params = { 0 };
-        EXPECT_FAILURE_WITH_ERRNO(s2n_kem_recv_public_key(&io_stuffer, &kem_params), S2N_ERR_NULL);
-
-        kem_params.kem = &s2n_test_kem;
-
-        /* The given public key length doesn't match the KEM's actual public key length */
-        DEFER_CLEANUP(struct s2n_blob io_blob_3 = { 0 }, s2n_free);
-        EXPECT_SUCCESS(s2n_alloc(&io_blob_3, 5));
-        struct s2n_stuffer io_stuffer_3 = { 0 };
-        EXPECT_SUCCESS(s2n_stuffer_init(&io_stuffer_3, &io_blob_3));
-        uint8_t bad_pk_input_3[] = { 0, 3, 3, 3, 3 };
-        EXPECT_SUCCESS(s2n_stuffer_write_bytes(&io_stuffer_3, bad_pk_input_3, 5));
-        EXPECT_SUCCESS(s2n_stuffer_reread(&io_stuffer_3));
-        EXPECT_FAILURE_WITH_ERRNO(s2n_kem_recv_public_key(&io_stuffer_3, &kem_params), S2N_ERR_BAD_MESSAGE);
-    };
-    {
-        /* Happy case(s) for s2n_get_kem_from_extension_id() */
-
-        /* The kem_extensions and kems arrays should be kept in sync with each other */
-        kem_extension_size kem_extensions[] = {
-            TLS_PQ_KEM_EXTENSION_ID_KYBER_512_R3
+    for (int len_prefixed = 0; len_prefixed < 2; len_prefixed++) {
```

Comment:
Some of these tests don't actually use the len_prefixed variable, maybe take those out of the loop?

Reply:
Done.